day2

Nitesh292003 opened the pull request:
![Screenshot from 2025-04-03 16-22-30](https://github.com/user-attachments/assets/41ece538-7796-4b6d-ab8d-1a1c87b8d1e0)
![Screenshot from 2025-04-03 16-22-44](https://github.com/user-attachments/assets/c18969a7-2bf6-4127-83ef-d9b9ba0f4730)

Nitesh292003 committed:
ques1

Nitesh292003 committed:
ques2

Nitesh292003 committed:
ques3

Nitesh292003 committed:
question4

Nitesh292003 committed:
question5

Nitesh292003 committed:
ques6

joshpratik commented on nitesh/question1.txt:
```diff
@@ -0,0 +1,76 @@
+Q1. What is cherry-pick? Explain the use case of it and also write steps to do it.
```
avoid copy pasting from AI

joshpratik reviewed (commented):
add proper PR title

joshpratik reviewed (commented):
add folder name as day2

Nitesh292003 committed:
changes done

Nitesh292003 commented:
combined 3 commits
![Screenshot from 2025-04-08 16-09-12](https://github.com/user-attachments/assets/69bc8b3a-bbe9-4d6a-8607-c1a8c3bf3919)
![Screenshot from 2025-04-08 16-06-35](https://github.com/user-attachments/assets/46e02ca1-457f-4a46-8fee-1c4039684c5e)
![Screenshot from 2025-04-08 16-05-57](https://github.com/user-attachments/assets/c8f603c7-1467-4e46-b2d5-c1dea7e89509)

Nitesh292003 commented:
deleted the commit using rebase -
![Screenshot from 2025-04-08 16-15-25](https://github.com/user-attachments/assets/c712b44f-4d0d-48d2-ad6a-ba3e01d7bbfe)
deleted the commit using reset hard-
![Screenshot from 2025-04-08 16-20-43](https://github.com/user-attachments/assets/d71516a3-acd5-4dfa-8dd0-ada87315d4cf)
![Screenshot from 2025-04-08 16-20-24](https://github.com/user-attachments/assets/3932479c-9fa8-45cd-b096-5e94405eb47a)

Nitesh292003 commented:
push using force-
![Screenshot from 2025-04-08 17-02-15](https://github.com/user-attachments/assets/542a6965-a10e-4fa0-b323-413a04876def)
push using force-with-lease-
![Screenshot from 2025-04-08 17-08-32](https://github.com/user-attachments/assets/c46babc3-fa05-45e7-8179-e33ed4628277)

Nitesh292003 commented:
set multipe remote repositories-
![Screenshot from 2025-04-08 17-17-57](https://github.com/user-attachments/assets/007a8282-113b-41ab-bea4-077b07ba004b)
